incompatibility/incompatibility-report/reporter.go

```diff
@@ -194,7 +248,118 @@ func schemaSearch(schema *openapiv3.Schema, keys []string) []*Incompatibility {
+	}
+	if schema.MinItems != 0 {
+		incompatibilities = append(incompatibilities,
+			newIncompatibility(Severity_WARNING, IncompatibiltiyClassification_DataValidation, addKeyPath(keys, "minItems")...))
+	}
+	if schema.UniqueItems {
+		incompatibilities = append(incompatibilities,
+			newIncompatibility(Severity_WARNING, IncompatibiltiyClassification_DataValidation, addKeyPath(keys, "uniqueItems")...))
+	}
+	if len(schema.AllOf) != 0 {
+		incompatibilities = append(incompatibilities,
+			newIncompatibility(Severity_FAIL, IncompatibiltiyClassification_Inheritance, addKeyPath(keys, "allOf")...))
+	}
+	if len(schema.OneOf) != 0 {
+		incompatibilities = append(incompatibilities,
+			newIncompatibility(Severity_FAIL, IncompatibiltiyClassification_Inheritance, addKeyPath(keys, "oneOf")...))
+	}
+	if len(schema.AnyOf) != 0 {
+		incompatibilities = append(incompatibilities,
+			newIncompatibility(Severity_FAIL, IncompatibiltiyClassification_Inheritance, addKeyPath(keys, "anyOf")...))
+	}
+	if schema.Items != nil {
+		for ind, item := range schema.Items.SchemaOrReference {
+			incompatibilities = append(incompatibilities, schemaSearch(item.GetSchema(), addKeyPath(keys, "items", strconv.Itoa(ind)))...)
+		}
+	}
+	if schema.Properties != nil {
+		for ind, prop := range schema.Properties.AdditionalProperties {
+			incompatibilities = append(incompatibilities, schemaSearch(prop.Value.GetSchema(), addKeyPath(keys, "properties", strconv.Itoa(ind), prop.Name))...)
+		}
+	}
+	if schema.AdditionalProperties != nil && schema.AdditionalProperties.GetSchemaOrReference() != nil {
+		incompatibilities = append(incompatibilities,
+			schemaSearch(schema.AdditionalProperties.GetSchemaOrReference().GetSchema(), addKeyPath(keys, "additionalProperties"))...)
+	}
+
+	return incompatibilities
+}
+
+// responseSearch scans for incompatibilities in a response object
+func responseSearch(resp *openapiv3.Response, keys []string) []*Incompatibility {
+	var incompatibilities []*Incompatibility
+	if resp == nil {
+		return incompatibilities
+	}
+	if resp.Headers != nil {
+		for _, prop := range resp.Headers.AdditionalProperties {
+			incompatibilities = append(incompatibilities,
+				headerSearch(prop.Name, prop.GetValue().GetHeader(), addKeyPath(keys, prop.Name))...,
+			)
+		}
+	}
+	if resp.Content != nil {
+		for _, prop := range resp.Content.AdditionalProperties {
+			incompatibilities = append(incompatibilities,
+				contentSearch(prop.Value, addKeyPath(keys, prop.Name))...,
+			)
+		}
+	}
+	return incompatibilities
+}
+
+//  headerSearch scans for incompatibilities in a header object
+func headerSearch(headerName string, header *openapiv3.Header, keys []string) []*Incompatibility {
+	var incompatibilities []*Incompatibility
+	if header == nil {
+		return incompatibilities
+	}
+	paramEquiv := header2Paramter(headerName, header)
+	incompatibilities = append(incompatibilities,
```

Comment:
Nice work here. I am sure it was tempting to inline this but adding the headerSearch as its own function does a good job of communicating the relation between headers and parameters